controllers/property_controller/property_address/updatePropertyAddress.js

```diff
@@ -0,0 +1,101 @@
+const { checkSchema } = require('express-validator/check');
+
+const PropertyAddress = require('../../../models/PropertyAddress');
+const { checkValidations, validPropertyAddress } = require('../../../utils/customValidations');
+
+module.exports = {
+    validation: checkSchema({
+        property_address_id: {
+            in: ['parmas', 'body'],
+            isString: true,
+            escape: true,
+            trim: true,
+            errorMessage: 'property_address_id is required',
+            custom: {
+                options: validPropertyAddress,
+            },
+        },
+        options: {
+            in: ['params', 'body'],
+            errorMessage: 'The options object is required',
+        },
+        'options.address_line_1': {
+            optional: true,
+            errorMessage: 'address_line_1 is required',
+            trim: true,
+            isString: true,
+            escape: true,
+            isLength: {
+                errorMessage: 'address_line_1 should be at least 3 characters long',
+                options: { min: 3 },
+            },
+        },
+        'options.address_line_2': {
+            optional: true,
+            errorMessage: 'address_line_2 must be a valid string',
+            trim: true,
+            isString: true,
+            escape: true,
+            isLength: {
+                errorMessage: 'address_line_2 should be at least 2 characters long',
+                options: { min: 2 },
+            },
+        },
+        'options.address_line_3': {
+            optional: true,
+            errorMessage: 'address_line_3 must be a valid string',
+            trim: true,
+            isString: true,
+            escape: true,
+            isLength: {
+                errorMessage: 'address_line_3 should be at least 1 characters long',
+                options: { min: 1 },
+            },
+        },
+        'options.name_city': {
+            optional: true,
+            errorMessage: 'name_city is required',
+            trim: true,
+            isString: true,
+            escape: true,
+            isLength: {
+                errorMessage: 'name_city should be at least 3 characters long',
+                options: { min: 3 },
+            },
+        },
+        'options.state': {
+            optional: true,
+            errorMessage: 'state is required',
+            trim: true,
+            isString: true,
+            escape: true,
+            isLength: {
+                errorMessage: 'state should be at least 2 characters long',
+                options: { min: 2 },
+            },
+        },
+        'options.zipcode': {
+            optional: true,
```

Comment:
I know optional here means they don't have to edit it; but does it also introduce a loophole where the user can edit and delete it?

Reply:
I see the error: Zipcode is required, but how does it know to pop that up and prevent a user from removing the text from the string field?